### ISSUE TITLE
Prompt storage is buggy, losing prompts

### ISSUE THREAD

**Issue by tarasglek** [tarasglek/chatcraft.org#219]
I posted on twitter:
![image](https://github.com/tarasglek/chatcraft.org/assets/857083/b8a24824-da4a-471d-81f1-e646ced6e2d5)

Then came back to this after few min:
<img width="1330" alt="image" src="https://github.com/tarasglek/chatcraft.org/assets/857083/f3c0c958-78bd-49a4-971f-2fab092b4095">

Pretty sure I did nothing in-between. Maybe i opened/closed chatcraft in another tab

**Comment by humphd** [comment]
Question: did you "Clear Messages"?  I would put money on this explanation, which would (currently) reset the system message too.  I think users likely wouldn't expect "Clear Messages" to also affect the system message.

I can do a PR to leave that alone.

Also, I need to take away "Clear Messages" from you, since it constantly causes you to lose your data.

**Comment by tarasglek** [comment]
no, you can see same messages in there as in my twitter, just the system prompted changed

**Comment by alexisargyris** [comment]
FYI, today I found out I lost my custom prompt. I don't believe it was due to any action on my side

**Comment by tarasglek** [comment]
https://github.com/tarasglek/chatcraft.org/assets/857083/218a98bc-4e53-41ec-b336-7080ab4749ff

repro

**Comment by humphd** [comment]
@tarasglek if you modify the prompt (which you did in that video), it updates in the db, thus changing the value of the starred prompt.

I don't think the code is buggy here, but it might not be working how people expect.  The values we show for favourite prompts are "live" in that we're querying the db to show you what you have.  If you modify them, the UI is reactive to the db, so they update.

@alexisargyris can you tell me more about your case?  How does it compare to what @tarasglek showed in his video?

**Comment by alexisargyris** [comment]
Thanks for your reply. I am a regular user, up to now had no problems at all. Today, I found out that my custom prompt had disappeared. Thought you made some kind of upgrade and lost custom prompts in the process. Not a big deal, just wanted to let you know.

BTW, thanks for ChatCraft, it's very, very useful, especially for being web-based (i.e., no special port required).

**Comment by humphd** [comment]
@alexisargyris thanks for your feedback and filing an issue. I suspect we *do* have bugs around this code, since it's so new and largely untested in the wild.  Feedback from regular users like you is amazing to get!

While you're here, consider adding some thoughts to https://github.com/tarasglek/chatcraft.org/issues/194.  Let us know what we should do next.

**Comment by tarasglek** [comment]
I disagree.
In my mind 
1) I wrote a prompt
2) Stared it to save it
3) then decided to switch to another prompt to maybe test it old vs new
4) now my prompt is gone

**Comment by humphd** [comment]
Makes sense.  So to do what you suggest, we have to separate the storage of saved/starred prompts from the original messages (i.e., add a table).

So let's say I star a system prompt:

- are you able to edit it?  If so, where?
- if you edit the system prompt you just starred (content is now different), does it then lose its star in the UI?  How do you find it again?
- how and where do you delete these?

I initially wanted to use the existing storage in order to not have to build a parallel system for managing system prompts.

**Comment by tarasglek** [comment]
I think we need to move to document-name based model. Eg everything has a name and it's clear what you are editing.

However in short term, staring could make prompts read only and feature an edit-as-new instead of edit

**Comment by kosty** [comment]
Had a similar experience to @tarasglek , yet while debugging the code found a way around the tool to "kinda" make it work. Creating a new blank chat to enter new prompt and starring it after edit seems to work consistently. Although creating numerous messages on DB. It feels like what @humphd  is saying is correct though. UI does not feel very intuitive, especially when it comes to creation of a custom prompts collection. 

There are few avenues I could think of:
* easier one, modify UI so it's clear what is being done, like change "new chat" to "new prompt" wording, disable "save" button by default, or just adding "Add"/"Update" dedicated buttons to System Prompt editing card.
* fancier one, move system prompts out of "messages" table to it's own dedicated table and it's dedicated management cycle.

What do you guys think?

**Comment by humphd** [comment]
I think the right solution here is to create a new table for the system prompts.  We can leave `messages` alone, and copy things over when you star, or remove when you un-star.

To do this, we need to write a migration so we don't lose the current data, but can bring it into the new table in an updated version.

@kosty are you willing to give this a shot?

**Comment by kosty** [comment]
absolutely, have tons of fun learning Dexie, Typescript, React with ChatCraft along the way

**Comment by humphd** [comment]
Great.  Let us know if you need guidance, and obviously, ask ChatCraft for help :)

**Comment by kosty** [comment]
@humphd one of the “uggghhhhh” sides of choosing text as id for prompt is that in order to add this line #333   to every non-default prompt one would:
* open existing non-default prompt 
* add a line
* save
* star
* find old non-default prompt (not easy given truncation for UI dropdown)
* unstar

feels a bit too much, don’t you think? 🤔

**Comment by tarasglek** [comment]
Man if we had normal sql, this would be trivial
